### PR TITLE
AI - Small refactoring based on previous feedback on the multiple

### PR DESCRIPTION
**Pull request by ron-murhammer**
resource support enhancement

[Review on Reviewable](https://reviewable.io/reviews/triplea-game/triplea/206)

### CODE REVIEW

File: src/games/strategy/triplea/ai/proAI/ProAI.java [hunk 1]
```diff
@@ -121,11 +121,11 @@ public ProAI(final String name, final String type) {
 
   public static void Initialize(final TripleAFrame frame) {
     LogUI.initialize(frame); // Must be done first
-    LogUtils.log(Level.FINE, "Initialized Hard AI");
+    ProLogUtils.log(Level.FINE, "Initialized Hard AI");
```
Comment (DanVanAtta) [inline src/games/strategy/triplea/ai/proAI/ProAI.java]: Violation of DRY having "Level.FINE" repeated many times. I recommend creating another method signature for log in ProLogUtils that takes just a string and defaults to Level.FINE.

**Comment by DanVanAtta** [comment]
Looks like there is only a class name change here. It's an improvement  :shipit:

File: src/games/strategy/triplea/ai/proAI/ProAI.java (hunk 1 above)
Comment (DanVanAtta) [inline src/games/strategy/triplea/ai/proAI/ProAI.java]: Specifically, so then this line could look like: `ProLogUtils.log("Initialized Hard AI")`

**Comment by DanVanAtta** [comment]
TBH, ProLogUtils could be improved as a name. ProLogger I think is slightly better as suggestion. There is not as much utility functions there as it is an actually logger.

**Comment by DanVanAtta** [comment]
This is a bit of a pet peeve of mine, so, my best efforts to not rant.

But please avoid this type of statement in your commit comments: "based on previous feedback"

If someone is reviewing the history, that is almost a big FU to them. The "previous feedback" is in some long gone pull request and is not in the commit history. Meanwhile that person is reviewing each code change in the history perhaps trying to figure out why some odd behavior is happening.

In addition, it actually made *this* review more difficult. I could have more easily reviewed it checking that it *only* contained the class name change, rather than having to figure that out. 

So bottom line, please have it so your commit comment always describes what changed, and avoid the commit comment that only states: "incorporated pull request feedback." It's an inappropriate comment because the assumed context is too narrow in scope.

**Comment by ron-murhammer** [comment]
So cleaned up the logging classes/methods some more. I actually did a few of the other changes from the previous review such as rename to 'purchaseOption' and rename ResourceData to ResourceTracker. May be a few other small things. The logging name makes it hard to see anything else though.
